### PR TITLE
Pre-mangle voucher data to clean spaces before we run the validations.

### PR DESCRIPTION
**Pull request by charlesstrange2**
https://trello.com/c/P8hFB5ET/1245-remimplement-hotfix-patch-vouchers-with-spaces-in-the-code-are-confusing-the-system-vvv-ee

### CODE REVIEW

File: app/Http/Requests/StoreAppendBundleRequest.php [hunk 1]
```diff
@@ -3,6 +3,7 @@
 namespace App\Http\Requests;
 
 use Illuminate\Foundation\Http\FormRequest;
+use App\Voucher;
```
Comment (katjam) [inline app/Http/Requests/StoreAppendBundleRequest.php]: :smile: You know me - I like these to be ABC - but not stopping you. Hardly relevant with 2 imports!

File: app/Http/Requests/StoreAppendBundleRequest.php [hunk 2]
```diff
@@ -37,4 +38,21 @@ public function rules()
...
         return $rules;
     }
+
+    protected function prepareForValidation()
+    {
+        // get the input and remove null/empty values.
+        // TODO: 5.5 upgrade will need to change this to 'all' with params.
+        $input = array_filter(
+            $this->only(['start', 'end']),
+            'strlen'
```
Comment (katjam) [inline app/Http/Requests/StoreAppendBundleRequest.php]: Not too keen on 0 == false but I think it's OK here.

File: tests/Unit/Controllers/Store/BundleControllerTest.php [hunk 3]
```diff
@@ -397,4 +397,32 @@ public function testICannotDisburseAnEmptyBundle()
             ->seePageIs($route)
             ->assertResponseStatus(200);
     }
+
+    /** @test */
+    public function itCanAcceptAndCleanVouchersWithSpacesIn()
```
Comment (katjam) [inline tests/Unit/Controllers/Store/BundleControllerTest.php]: Do you also want to test that `null` and `""` values will get cleared from the list?

File: app/Http/Requests/StoreAppendBundleRequest.php (hunk 2 above)
Comment (charlesstrange2) [reply app/Http/Requests/StoreAppendBundleRequest.php]: it's apparently a well known array_filter trick :man_shrugging:

File: tests/Unit/Controllers/Store/BundleControllerTest.php [hunk 4]
```diff
@@ -397,4 +397,79 @@ public function testICannotDisburseAnEmptyBundle()
...
+        /** @var Bundle $currentBundle */
+        // Get our currentBundle
+        $currentBundle = $this->registration->currentBundle();
+
+        // See that it's got one voucher.
+        $this->assertEquals(1, $currentBundle->vouchers()->count());
+    }
+
+    /** @test */
+    public function itHasSparseFormDataCleanedBeforeProcessing()
```
Comment (katjam) [inline tests/Unit/Controllers/Store/BundleControllerTest.php]: Title is a bit opaque - but otherwise. :+1:
